units/systemd-fsck-root.service.in

```diff
@@ -11,7 +11,8 @@
 Description=File System Check on Root Device
 Documentation=man:systemd-fsck-root.service(8)
 DefaultDependencies=no
-Conflicts=shutdown.target
+IgnoreOnIsolate=yes
```

Comment:
why this? poweroff is enqueued with job mode "replace-irreversibly", not with "isolate", hence why?

Reply:
Hmm, I think this makes sense on its own. We don't want `isolate` to bring down mounts either.

Reply:
yes, that was my thinking too

Reply:
that should not be necessary, though? i mean, isolation just stops the unneeded stuff, i.e. the stuff that has no dep anymore in dep tree of the new unit we are isolating to. hence, in particular system-fsck-root.service should never be considered for stopping anyway given that the unit we are isolating to will always require the root fs to stick around.

hence, i'd really drop that, i don't see the benefit.

Reply:
it's actually needed, otherwise it is pulled down on soft-reboot, bringing the disks down too

Reply:
can you explain? soft-reboot doesn't do isolation, does it?

Reply:
It does, that's why IgnoreOnIsolate is needed everywhere

Reply:
> isolation just stops the unneeded stuff, i.e. the stuff that has no dep anymore in dep tree of the new unit we are isolating to

That's not exactly how isolate is implemented. We would skip a unit only if IgnoreOnIsolate= or when it has a job enqueued in the transaction already - which is different from ignoring still required units.

All mount units have IgnoreOnIsolate= set, which means they _and their deps_ don't get a job enqueued when isolating. Then, when `systemd-fsck@.service` gets checked, it gets a stop job enqueued, resulting in mounts depending on it getting brought down.

Reply:
src/systemctl/systemctl-start-unit.c says "start-irreversibly" is used

and so does execute_shutdown_or_sleep() in src/login/logind-dbus.c

hence, afaics both paths to soft-reboot do not go via isolation.

Reply:
> hence, afaics both paths to soft-reboot do not go via isolation.

They do. Just try it, there's the integration tests for it...

Reply:
if that indeed is the case then that should be fixed.

Reply:
> That's not exactly how isolate is implemented. We would skip a unit only if IgnoreOnIsolate= or when it has a job enqueued in the transaction already - which is different from ignoring still required units.
> 
> All mount units have IgnoreOnIsolate= set, which means they _and their deps_ don't get a job enqueued when isolating. Then, when `systemd-fsck@.service` gets checked, it gets a stop job enqueued, resulting in mounts depending on it getting brought down.

So I can accept this to some point, i.e. .mount units do Require=systemd-fsck@.service, hence yes, if we exclude mount units from isolation by default, we should ignore the fsck services they require, too.

but systemd-fsck-root.service is different, it's not actually rquired by the root mount, because it can actually run after the root mount for compat with crazy systems that have no initrd.

Reply:
Fine, I just added it to the root one to keep the settings mirrored, but I don't need actually need it changed for my use case, dropped